pkg/network/blockqueue_test.go

```diff
@@ -0,0 +1,71 @@
+package network
+
+import (
+	"testing"
+	"time"
+
+	"github.com/CityOfZion/neo-go/pkg/core"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBlockQueue(t *testing.T) {
+	chain := &testChain{}
+	// notice, it's not yet running
+	bq := newBlockQueue(0, chain)
+	blocks := make([]*core.Block, 11)
+	for i := 1; i < 11; i++ {
+		blocks[i] = &core.Block{BlockBase: core.BlockBase{Index: uint32(i)}}
+	}
+	// not the ones expected currently
+	for i := 3; i < 5; i++ {
+		bq.putBlock(blocks[i])
+	}
+	// nothing should be put into the blockchain
+	assert.Equal(t, uint32(0), chain.BlockHeight())
+	assert.Equal(t, 2, bq.length())
+	// now added expected ones (with duplicates)
+	for i := 1; i < 5; i++ {
+		bq.putBlock(blocks[i])
+	}
+	// but they're still not put into the blockchain, because bq isn't running
+	assert.Equal(t, uint32(0), chain.BlockHeight())
+	assert.Equal(t, 4, bq.length())
+	go bq.run()
+	// run() is asynchronous, so we need some kind of timeout anyway and this is the simplest one
+	for i := 0; i < 5; i++ {
+		if chain.BlockHeight() != 4 {
+			time.Sleep(time.Second)
```

Comment:
"I see dead people"
sleep...r u sure it's a good choice?)

Reply:
Well, you can make a `context` and set a timeout for it, but it doesn't look like that much of an improvement for me especially if you're to consider additional code complexity it brings. As written in the comment above, we have an asynchronous event that we're waiting for and we can't just block forever here because it's a test code, it can fail easily, so we need a reasonable timeout (actually Go has a timeout for tests, but that's 10 minutes, not something you'd like to spend waiting for unit testing results, at least not in our small project). And yes it does poll for the result, but again, it's testing code, it should be simple and can even be dumb sometimes.

Reply:
If there are really better suggestions --- I'm open for them.

Reply:
thought about some way of touching manually persist. But probably for that we have to have interface and several impls of BC. Do agree that for now it's not a big issue.